lib/rails/controller/testing/template_instrumentation.rb

```diff
@@ -0,0 +1,16 @@
+module Rails
+  module Controller
+    module Testing
+      module InstrumentDetermineTemplate
+        def determine_template(options)
+          super.tap do |found|
+            if found
+              instrument_payload = { template_identifier: found.identifier, virtual_path: found.virtual_path, options: options }
```

Comment:
```suggestion
              instrument_payload = { template_identifier: found.identifier, virtual_path: found.try(:virtual_path), options: options }
```

@cpruitt `virtual_path` isn't always available, like if you're rendering a string rather than a template. This should `try` to call `virtual_path` so it doesn't fail if the method is missing.

I reported this error here: https://github.com/rails/rails-controller-testing/pull/46#issuecomment-572729139

Reply:
@bbugh Thanks for the suggested change. Our test implementation changed such that I'm not in a great position to reproduce this problem or test this fix. I believe the original issue that this PR was intended to resolve changed and this PR just happened to still fix a similar problem. I'm not currently able to allocate time to get up to speed to verify that and test this out. If you have available time, I'd wholeheartedly endorse you taking this and running with it in a new PR. I'm happy to help answer any questions I can (thought I don't know how much help that would be). 🙂